packages/ra-ui-materialui/src/input/NumberInput.stories.tsx

```diff
@@ -25,6 +25,26 @@ export const Basic = () => (
     </AdminContext>
 );
 
+type Post = {
+    published: boolean;
+    views: number;
+    title: string;
+};
+
+export const Typed = () => (
+    <AdminContext>
+        <Create
+            resource="posts"
+            record={{ id: 123, views: 23 }}
+            sx={{ width: 600 }}
+        >
+            <SimpleForm>
+                <NumberInput<Post> source="views" />
```

Comment:
Since the change to `extends CommonInputProps<number>` I would have expected to only be able to use `"views"` as value for `source`, unfortunately all 3 fields are valid according to TS. Is it intentional?

Reply:
It is. Because of format/parse we decided to not add more constraints, a at least for now